Bottom: 1647bb2d69e06b5b0dd47de329909a9c1d995599
Top:    ec7b9dd033526504b45f904a8882cce9b1ce849c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-10 17:26:29 +0100

Refresh of fix-crash-hovering-over-unconn

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 346a6db..f8be7dd 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1968,17 +1968,23 @@ describe_location (LocationType X, LocationType Y)
   void *ptr1, *ptr2, *ptr3;
   int type;
   int Range = 0;
-  char *elename;
+  char *elename = "";
   char *pinname;
   char *netname = NULL;
   char *description;
 
   /* check if there are any pins or pads at that position */
 
-  type = SearchObjectByLocation (LOOKUP_FIRST,
+  type = SearchObjectByLocation (PIN_TYPE | PAD_TYPE,
                                  &ptr1, &ptr2, &ptr3, X, Y, Range);
   if (type == NO_TYPE)
-    type = SearchObjectByLocation (LOOKUP_MORE,
+    type = SearchObjectByLocation (VIA_TYPE | LINE_TYPE | ARC_TYPE,
+                                   &ptr1, &ptr2, &ptr3, X, Y, Range);
+  if (type == NO_TYPE)
+    type = SearchObjectByLocation (RATLINE_TYPE,
+                                   &ptr1, &ptr2, &ptr3, X, Y, Range);
+  if (type == NO_TYPE)
+    type = SearchObjectByLocation (POLYGON_TYPE,
                                    &ptr1, &ptr2, &ptr3, X, Y, Range);
 
   if (type == NO_TYPE)
@@ -1989,7 +1995,9 @@ describe_location (LocationType X, LocationType Y)
       GetLayerNumber (PCB->Data, (LayerTypePtr) ptr1) >= max_layer)
     return NULL;
 
-  elename = "DUMMY"; // UNKNOWN (NAMEONPCB_NAME ((ElementTypePtr) ptr1));
+  if (type == PIN_TYPE || type == PAD_TYPE)
+    elename = UNKNOWN (NAMEONPCB_NAME ((ElementTypePtr) ptr1));
+
   pinname = ConnectionName (type, ptr1, ptr2);
 
   if (pinname == NULL)
